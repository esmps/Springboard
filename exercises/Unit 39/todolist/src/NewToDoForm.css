.newToDoForm{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 20px;
    margin: 20px;
}
.newToDoForm-textarea{
    width: 400px;
    height: 20px;
    font-size: 18px;
}
.newToDoForm-textareaLabel{
    font-size: 20px;
}