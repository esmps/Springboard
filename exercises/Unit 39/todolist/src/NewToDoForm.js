import {useState} from "react";
import { v4 as uuid } from 'uuid';
import './NewToDoForm.css'

function NewToDoForm({newToDo}){
    const [formData, setFormData] = useState({toDoText: ""});

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(formData => ({
          ...formData,
          [name]: value
        }));
    };

    const gatherInput = evt => {
        evt.preventDefault();
        if (formData.toDoText){
            newToDo({...formData, id: uuid()})
            setFormData({toDoText: ""});
        }
    }

    return (
        <form className="newToDoForm" onSubmit={gatherInput}>
            <div>
                <label 
                    className="newToDoForm-textareaLabel"
                    htmlFor="toDoText">
                        What do you need to do?
                </label>
                <br/>
                <textarea
                    onChange={handleChange}
                    name="toDoText"
                    id="toDoText"
                    className="newToDoForm-textarea"
                    value={formData.toDoText}
                >
                </textarea>
                
            </div>
            <button className="newToDoForm-btn" id="btn-newToDo">Add!</button>
        </form>
    )

}

export default NewToDoForm;