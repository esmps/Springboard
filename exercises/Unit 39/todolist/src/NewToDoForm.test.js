import { render } from "@testing-library/react";
import NewToDoForm from "./NewToDoForm";

//smoke test
it("renders without crashing", function() {
  render(<NewToDoForm />);
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<NewToDoForm />);
  expect(asFragment()).toMatchSnapshot();
});