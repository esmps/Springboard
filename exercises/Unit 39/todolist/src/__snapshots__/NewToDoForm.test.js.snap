// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="newToDoForm"
  >
    <div>
      <label
        class="newToDoForm-textareaLabel"
        for="toDoText"
      >
        What do you need to do?
      </label>
      <br />
      <textarea
        class="newToDoForm-textarea"
        id="toDoText"
        name="toDoText"
      />
    </div>
    <button
      class="newToDoForm-btn"
      id="btn-newToDo"
    >
      Add!
    </button>
  </form>
</DocumentFragment>
`;
